cpp/CMakeLists.txt

```diff
@@ -339,16 +355,30 @@ message(STATUS "CUDF_TEST_LIST set to: ${CUDF_TEST_LIST}")
 # - add google benchmark --------------------------------------------------------------------------
 
 if(BUILD_BENCHMARKS)
+    find_package(benchmark)
+    
+    if(GBENCH_ROOT)
+        message(STATUS "Google C++ Benchmarking Framework (Google Benchmark) found in ${GBENCH_ROOT}")
+        add_subdirectory(${CMAKE_SOURCE_DIR}/benchmarks)
```

Comment:
So instead of `GBENCH_ROOT`, it will search in `benchmark_ROOT` in the `find_package` call. If the package is found it will then set a `benchmark_FOUND` variable, along with variables for the library / include dirs (unsure exactly what they will be named).

So instead of checking `GBENCH_ROOT` here, we can do something along the lines of `if(NOT benchmark_FOUND)`  to indicate we need to build it from source using our configuration module.

Reply:
Right, which is done below so I have checks for it after it installs as well. But if someone does not have a conda (or other) install and I check for GBENCH/GTEST_FOUND instead of GBENCH_ROOT before install it fails out. So in this configuration, it checks for the root, if found, continues. If not found, installs, sets up includes and some vars (definitely required in this pattern) then checks for GTEST_FOUND at the bottom in case the install failed, if failed in the install step for some reason disable benchmarks or tests so there are not conflicts with other steps.